### PR TITLE
Finalise Login Page with Text Field Validation

### PR DESCRIPTION
**Pull request by s-hane**
Default:

image

On Focus:
image

### CODE REVIEW

File: src/components/Button/index.styles.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ const StyledButton = styled(Button)<StyledButtonProps>`
     padding-right: 48px;
     padding-top: 12px;
     padding-bottom: 12px;
-    border-radius: 8px
+    border-radius: 8px;
+    margin: 8px;
```
Comment (ericgmoon) [inline src/components/Button/index.styles.ts:17]: Should there really be a margin built in to the button? I feel like margin is something that should be added when USING the button.

Comment (ericgmoon) [inline src/components/Button/index.styles.ts:17]: Should there really be a margin built in to the button? I feel like margin is something that should be added when USING the button.

File: src/components/Login/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+
+import Login from '.';
+
+export default {
+  title: 'Components/Login',
```
Comment (ericgmoon) [inline src/components/Login/index.stories.tsx:8]: This is a page, so should be in the `Pages/` folder

File: src/components/Login/index.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+import { useForm } from 'react-hook-form';
+
+import Button from '../Button';
+import TextField from '../TextField';
+
+import { StyledLogin } from './index.styles';
+
+// interface LoginProps {
```
Comment (ericgmoon) [inline src/components/Login/index.tsx:10]: Unused code - remove

File: src/components/Login/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+const Login = (/* { mobile = false }: LoginProps */) => {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const onSubmit = (data: Data) => console.log(data);
+  console.log(errors);
+
+  return (
+    <StyledLogin onSubmit={handleSubmit(onSubmit)}>
+      <TextField placeholder="Email" {...register('email', { required: true, pattern: /^\S+@\S+$/i })} />
```
Comment (ericgmoon) [inline src/components/Login/index.tsx:26]: I feel like we should move the contents of the register out, as more complex forms will result in the DOM tree getting less readable.

**Review by ericgmoon — CHANGES_REQUESTED** [review]

**Comment by s-hane** [comment]
image

File: src/components/Pages/LoginPage/index.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+import { ComponentMeta, ComponentStory } from '@storybook/react';
+
+import LoginPage from '.';
```
Comment (ericgmoon) [inline src/components/Pages/LoginPage/index.stories.tsx:5]: Rename page to `SignInPage`?

File: src/components/Login/index.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+import {
+  StyledButton, StyledLogin, TextFieldContainer,
+} from './index.styles';
+
+interface Data {
+  email:string;
+  password:string;
+}
+
+const Login = (/* { mobile = false }: LoginProps */) => {
```
Comment (ericgmoon) [inline src/components/Login/index.tsx:16]: Rename to `SignIn`?

File: src/components/Pages/LoginPage/index.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+export default {
+  title: 'Pages/LoginPage',
+  component: LoginPage,
+} as ComponentMeta<typeof LoginPage>;
+
+const Template: ComponentStory<typeof LoginPage> = () => <LoginPage />;
+
+export const Default = Template.bind({});
+Default.args = {
```
Comment (ericgmoon) [inline src/components/Pages/LoginPage/index.stories.tsx:15]: I don't see any use of this prop, `placeholder` in the component. Is it meant to be here?

File: src/components/Login/index.styles.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import Button from '../Button';
+
+const StyledLogin = styled.form`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+`;
+
+const TextFieldContainer = styled.div`
+        margin: 12px;
```
Comment (ericgmoon) [inline src/components/Login/index.styles.ts:12]: Incorrect margin here?

File: src/components/Login/index.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const onSubmit = (data: Data) => console.log(data);
+  console.log(errors);
+
+  const getErrorMessage = (type: string, field: string): string => {
+    let error;
+
+    if (type === 'pattern') {
```
Comment (ericgmoon) [inline src/components/Login/index.tsx:24]: switch-case feels more appropriate here, as we'll be adding on more error messages over time

File: src/components/Login/index.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+  password:string;
+}
+
+const Login = (/* { mobile = false }: LoginProps */) => {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const onSubmit = (data: Data) => console.log(data);
+  console.log(errors);
+
+  const getErrorMessage = (type: string, field: string): string => {
+    let error;
```
Comment (ericgmoon) [inline src/components/Login/index.tsx:22]: The `error` variable is unnecessary. We could just return within the body of each if statement.

File: src/theme.ts [hunk 11]
```diff
@@ -26,6 +26,18 @@ const theme = {
...
       contrastText: '#fff',
     },
+    gray: {
+      main: '#EFEFEF',
+      dark: '#D8D8D8',
+      light: '#FBFBFB',
+      contrastText: '#393939',
+    },
+    darkGray: {
+      main: '#808080',
```
[the remark this answers is not in the document]
Comment (s-hane) [reply src/theme.ts:36]: This is copied straight from our figma idk what you mean by dark child

File: src/components/Login/index.tsx (hunk 6 above)
Comment (s-hane) [reply src/components/Login/index.tsx:16]: That’s been done

File: src/components/Login/index.styles.ts (hunk 8 above)
Comment (s-hane) [reply src/components/Login/index.styles.ts:12]: Why?

File: src/components/Login/index.tsx (hunk 9 above)
Comment (s-hane) [reply src/components/Login/index.tsx:24]: Okay

File: src/components/Login/index.tsx (hunk 10 above)
Comment (s-hane) [reply src/components/Login/index.tsx:22]: Yep I changed this already haven’t put it through yet

File: src/components/Login/index.tsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+interface Data {
+  email:string;
+  password:string;
+}
+
+const Login = (/* { mobile = false }: LoginProps */) => {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const onSubmit = (data: Data) => console.log(data);
+  console.log(errors);
```
[the remark this answers is not in the document]
Comment (s-hane) [reply src/components/Login/index.tsx:19]: Ok

File: src/components/Pages/LoginPage/index.tsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+import Login from '../../Login';
+
+import {
+  Bold, Container, Logo, RootContainer, Title,
+} from './index.styles';
+
+const LoginPage = () => {
+  const theme = useTheme();
+
+  const isMd = useMediaQuery(theme.breakpoints.up('md'));
```
[the remark this answers is not in the document]
Comment (s-hane) [reply src/components/Pages/LoginPage/index.tsx:20]: Checks screen size and changes variables accordingly

File: src/components/Pages/LoginPage/index.tsx [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+  const getOrientation = () : ('vertical' | 'horizontal' | undefined) => {
```
[the remark this answers is not in the document]
Comment (s-hane) [reply src/components/Pages/LoginPage/index.tsx:22]: No

File: src/components/Login/index.styles.ts (hunk 8 above)
Comment (ericgmoon) [reply src/components/Login/index.styles.ts:12]: As in, the line indent/file margin is wrong (compare to the indent above)

File: src/components/Pages/LoginPage/index.tsx (hunk 13 above)
[the remark this answers is not in the document]
Comment (ericgmoon) [reply src/components/Pages/LoginPage/index.tsx:20]: Not sure if `isMd` is the most suitable name

[the remark this answers is not in the document]
Comment (s-hane) [reply src/components/Pages/LoginPage/index.tsx:20]: Its checking if screen size is Md <- which is an MUI specified name

File: src/auth/index.ts [hunk 15]
```diff
@@ -80,7 +80,7 @@ export const resendConfirmationCode = async (username: string | undefined) =>
  * @param onSuccess
  * @param onFailure
  */
-export const signIn = async (
+export const SignIn = async (
```
Comment (ericgmoon) [inline src/auth/index.ts:83]: Wrong case

File: src/stories/Header.tsx [hunk 16]
```diff
@@ -5,12 +5,12 @@ import './header.css';
...
 interface HeaderProps {
   user?: {};
-  onLogin: () => void;
+  onSignIn: () => void;
   onLogout: () => void;
   onCreateAccount: () => void;
 }
 
-export const Header = ({ user, onLogin, onLogout, onCreateAccount }: HeaderProps) => (
+export const Header = ({ user, onSignIn, onLogout, onCreateAccount }: HeaderProps) => (
```
Comment (ericgmoon) [inline src/stories/Header.tsx:13]: Undo changes to stories folder

**Review by ericgmoon — CHANGES_REQUESTED** [review]

**Review by ericgmoon — APPROVED** [review]

**Pull request merged**